streams/src/main/java/org/apache/kafka/streams/processor/internals/RecordQueue.java

```diff
@@ -240,6 +240,7 @@ private void updateHead() {
                         deserialized.topic(), deserialized.partition(), deserialized.offset(), timestamp, timestampExtractor.getClass().getCanonicalName()
                 );
                 droppedRecordsSensor.record();
+                lastCorruptedRecord = raw;
```

Comment:
This is the actual fix -- we track the record with invalid ts as "corrupted record" instead of dropping it completely. The `CorrecuptedRecrods` allows us to update the offset later, but would be dropped by itself and not be processed.